lib/absinthe/phase/error.ex

```diff
@@ -15,18 +16,21 @@ defmodule Absinthe.Phase.Error do
     message: String.t,
     phase: module,
     locations: [loc_t],
+    extra: Keyword.t
   }
 
   @doc """
   Generate a phase error that relates to a specific point in the
   document.
   """
   @spec new(Absinthe.Phase.t, String.t, loc_t | [loc_t]) :: t
-  def new(phase, message, location) do
+  @spec new(Absinthe.Phase.t, String.t, loc_t | [loc_t], Keyword.t) :: t
+  def new(phase, message, location, extra \\ []) do
     %__MODULE__{
       phase: phase,
       message: message,
-      locations: List.wrap(location)
+      locations: List.wrap(location),
+      extra: Enum.filter(extra, &filter_extra_item/1)
```

Comment:
Do you think of a better place to enforce that ? Do you think passing a bad extra should cause some sort of error instead of just dropping the extra field ?